src/pkg/runner/actions.go

```diff
@@ -23,23 +23,38 @@ import (
 )
 
 func (r *Runner) performAction(action types.Action) error {
+
+	message.SLog.Debug(fmt.Sprintf("Action conditional is %s", action.If))
+
+	action, _ = utils.TemplateTaskActions(nil, action, action.With, r.variableConfig.GetSetVariables())
+	if action.If == "false" && action.TaskReference != "" {
+		message.SLog.Info(fmt.Sprintf("Skipping action %s", action.TaskReference))
+		return nil
+	} else if action.If == "false" && action.Description != "" {
+		message.SLog.Info(fmt.Sprintf("Skipping action %s", action.Description))
+		return nil
+	} else if action.If == "false" && action.Cmd != "" {
+		cmdEscaped := helpers.Truncate(action.Cmd, 60, false)
+		message.SLog.Info(fmt.Sprintf("Skipping action \"%s\"", cmdEscaped))
+		return nil
+	}
+
 	if action.TaskReference != "" {
 		// todo: much of this logic is duplicated in Run, consider refactoring
 		referencedTask, err := r.getTask(action.TaskReference)
 		if err != nil {
 			return err
 		}
-
-		// template the withs with variables
 		for k, v := range action.With {
 			action.With[k] = utils.TemplateString(r.variableConfig.GetSetVariables(), v)
 		}
+		for k, v := range referencedTask.Actions {
```

Comment:
It would be nice if we tracked variables in a `context.Context` object and had an API for evaluating (potentially) templated string values like so:

```go
interface Variable {
  Default() interface{}
  Value(context.Context) interface{} // interpolated value
  String() string // original expression
  IsExpression() bool // `true` if raw value contains `${{ ... }}` syntax
}
```

I think this interface could work for task/action inputs, environment variables, and eventually outputs.

```go
// before
runner.Run(taskFile, "default", setVariables)

// internally `runner.Run(...)` mutates `action` with templated values:
action, err := utils.TemplateTaskActions(nil, action, action.With, r.variableConfig.GetSetVariables())
if action.If == "true" {
  // ...
}
```

```go
// after
ctx := context.Background()
run := runner.New(tasksFile, runner.WithContext(ctx), runner.WithEnv(os.Getenv))

task := run.Task("default", inputs) // run task named `default`
task.Outputs() // => map[string]Variable

// internally `run.Task(...)` etc uses API like:
if action.If.Value(ctx) == "true" {
  // ...
}
```

This is basically what I was trying to do in #117 (see [`runner.Resolve(taskName)`](https://github.com/defenseunicorns/maru-runner/pull/117/files#diff-9dbd929135e99a9caedcd10f7e558826554f2c7c9b5848adb3574ae722a07101R154-R159)) but it's a very difficult refactor.

Reply:
It would be nice but likely a larger refactor - could target for a tech-debt issue if you wanted to create one.

Reply:
Appreciate the feedback @marshall007. In the interest of our immediate priorities though have to agree with @Racer159. Good tech debt issue or target of that elusive 2.0 we discussed previously that is more fully featured.